packages/ember-data/tests/integration/relationships/has_many_test.js

```diff
@@ -144,6 +144,52 @@ test("A serializer can materialize a hasMany as an opaque token that can be lazi
   });
 });
 
+test("If has-many options buildUrl is set, the adapter's buildURLForHasMany is called and the result is passed to findHasMany", function() {
+  Post.reopen({
+    comments: DS.hasMany('comment', { async: true, buildURL: true })
```

Comment:
The model layer shouldn't have any knowledge of whether we want to buildURLs or not